File: DaySeven/menstrualWahala/MenstrualApp.java
import java.util.Scanner;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class MenstrualApp {

     public static void main(String [] args) {

     Scanner scanner = new Scanner(System.in);

     System.out.println("Welcome to Menstrual App");

     System.out.print("Kindly enter your average cycle length: ");
     int length = scanner.nextInt();

     scanner.nextLine();

     System.out.print("Kindly enter the start date of your last period in this format (MM/dd/yyyy): ");
     String lastPeriodDate = scanner.nextLine();

     DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MM/dd/yyyy");

     LocalDate lastPeriodDates = LocalDate.parse(lastPeriodDate, formatter);

     LocalDate presentDate = LocalDate.now();

     int averageCycleLength = length;

     LocalDate nextPeriodDate = lastPeriodDates.plusDays(averageCycleLength);

     LocalDate ovulationDate = lastPeriodDates.plusDays(14);

     LocalDate safePeriodStart = ovulationDate.plusDays(1);

     LocalDate safePeriodEnd = nextPeriodDate.minusDays(1);

     LocalDate fertileWindowStart = lastPeriodDates.plusDays(10);

     LocalDate fertileWindowEnd = ovulationDate;

     System.out.println("Your next period date is: " + nextPeriodDate);

     System.out.println("Your ovulation date is: " + ovulationDate);

     System.out.println("Your safe period is from: " + safePeriodStart + " to " + safePeriodEnd);

     System.out.println("You are most likely to get pregnant within this duration: " + fertileWindowStart + " to " + fertileWindowEnd);


    }


}

File: DaySeven/menstrualWahala/menstrual_app.py
from datetime import datetime, timedelta

print("Hi sweethearts:), Welcome to Menstrual App")

name = input("kindly enter your name: ")

length = int(input("Kindly enter your average cycle length: "))

last_period_date = input("Kindly enter the start date of your last period in this format (MM/dd/yyyy): ")
last_period_dates = datetime.strptime(last_period_date, "%m/%d/%Y")

present_date = datetime.now()

average_cycle_length = length

next_period_date = last_period_dates + timedelta(days=average_cycle_length)

ovulation_date = next_period_date - timedelta(days=14)

safe_period_start = ovulation_date + timedelta(days=1)

safe_period_end = next_period_date - timedelta(days=1)

fertile_window_start = last_period_dates + timedelta(days=10)

fertile_window_end = ovulation_date

print(f"{name} your next period date is:", next_period_date.strftime("%m/%d/%Y"))

print(f"{name} your ovulation date is:", ovulation_date.strftime("%m/%d/%Y"))

print(f"{name} your safe period is from:", safe_period_start.strftime("%m/%d/%Y"), "to", safe_period_end.strftime("%m/%d/%Y"))

print(f"{name} you are most likely to get pregnant within this duration:", fertile_window_start.strftime("%m/%d/%Y"), "to", fertile_window_end.strftime("%m/%d/%Y"))